internal/v5/resources_test.go

```diff
@@ -461,3 +461,160 @@ func (s *ResourceSuite) TestListResourceWithBundle(c *gc.C) {
 		},
 	})
 }
+
+func (s *ResourceSuite) TestMetaResourcesSingleBadResourceId(c *gc.C) {
+	id := newResolvedURL("~charmers/precise/wordpress-0", -1)
+	s.addPublicCharm(c, storetesting.NewCharm(nil), id)
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler:      s.srv,
+		URL:          storeURL(id.URL.Path() + "/meta/resources/foo/bar"),
+		ExpectStatus: http.StatusNotFound,
+		ExpectBody: params.Error{
+			Code:    params.ErrMetadataNotFound,
+			Message: string(params.ErrMetadataNotFound),
+		},
+	})
+}
+
+func (s *ResourceSuite) TestMetaResourcesSingleResourceWithNegativeRevisionNumber(c *gc.C) {
+	id := newResolvedURL("~charmers/precise/wordpress-0", -1)
+	s.addPublicCharm(c, storetesting.NewCharm(nil), id)
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler:      s.srv,
+		URL:          storeURL(id.URL.Path() + "/meta/resources/foo/-2"),
+		ExpectStatus: http.StatusNotFound,
+		ExpectBody: params.Error{
+			Code:    params.ErrMetadataNotFound,
+			Message: string(params.ErrMetadataNotFound),
+		},
+	})
+}
+
+func (s *ResourceSuite) TestMetaResourcesSingleWithBundle(c *gc.C) {
+	id, _ := s.addPublicBundleFromRepo(c, "wordpress-simple", newResolvedURL("cs:~charmers/bundle/something-32", 32), true)
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler:      s.srv,
+		URL:          storeURL(id.URL.Path() + "/meta/resources/foo"),
+		ExpectStatus: http.StatusNotFound,
+		ExpectBody: params.Error{
+			Code:    params.ErrMetadataNotFound,
+			Message: string(params.ErrMetadataNotFound),
+		},
+	})
+}
+
+func (s *ResourceSuite) TestMetaResourcesSingleResourceNotInCharm(c *gc.C) {
+	id := newResolvedURL("~charmers/precise/wordpress-0", -1)
+	s.addPublicCharm(c, storetesting.NewCharm(nil), id)
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler:      s.srv,
+		URL:          storeURL(id.URL.Path() + "/meta/resources/foo"),
+		ExpectStatus: http.StatusNotFound,
+		ExpectBody: params.Error{
+			Code:    params.ErrMetadataNotFound,
+			Message: string(params.ErrMetadataNotFound),
+		},
+	})
+}
+
+func (s *ResourceSuite) TestMetaResourcesSingleResourceNotUploaded(c *gc.C) {
```

Comment:
After looking at the code, I see this is actually testing not even uploaded, a.k.a. the placeholder resource.